backend/src/main/java/com/example/backend/service/writing/WritingService.java

```diff
@@ -83,10 +85,54 @@ public WritingResponse getAndRecommendById(int id) {
     }
 
     public List<String> RecommendUsers(Integer curMemberId, Integer languageId) {
-        List<String> users = new ArrayList<>();
-        List<MemberLanguage> memberLanguages = memberLanguageRepository.get10ForWriting(languageId, curMemberId);
-        for (MemberLanguage m : memberLanguages) {
-            users.add(memberRepository.getOne(m.getMemberId()).getUsername());
+        ArrayList<String> users = new ArrayList<>();
+        HashSet<String> usernames = new HashSet<>(); //For quick lookup
+        HashMap<Integer, String> idUsernameMap = new HashMap<>();
+        //This probably needs to change when a lot of new members come
+        List<Member> members = memberRepository.findAll();
+        members.forEach(member -> idUsernameMap.put(member.getId(), member.getUsername()));
+        LocalDateTime curTime = LocalDateTime.now();
+        List<MemberLanguage> memberLanguages = memberLanguageRepository.findByLanguageIdExceptMemberIdOrderByLanguageLevelDesc(languageId, curMemberId);
+        int recommendationSize = Math.min(10, memberLanguages.size());
+        int qualifiedSize = recommendationSize*3/4;
+        int i = 0;
+        //Add active and well qualified members that are not busy and not recently assigned a task
```

Comment:
Maybe this logic can be divided into sub-methods so that readability can be increased. I couldn't see anything wrong with the logic.